extension/src/Settings/Settings.ts

```diff
@@ -55,14 +56,19 @@ export class Settings {
   public consoleLogOutput = false;
   public xliffCSVExportPath = "";
   public xliffCSVImportTargetState = "translated";
-  public loadSymbols = true;
+  public loadSymbols: string | boolean = DEPRECATED_VALUE;
   public refreshXlfAfterFindNextUntranslated = true;
   public enableTranslationsOnHover = true;
   public enableTelemetry = true;
   public enableTroubleshootingCommands = true;
   public useDictionaryInDTSImport = true;
   public enableXliffCache = true;
 
+  // new naming:
+  private _useSymbols = true;
+  public documentationOutputIndexFile = false;
+  public documentationOutputIndexFileDepth = 2;
```

Comment:
Could documentation be it's own class or interface? Having the object grouped and structured similar to package.json would be nice IMO. But is it practical?

Reply:
that would mean to access the property it would be like this:
```javascript
const settings = new Settings();
if (settings.documentation.outputIndexFile) {
     // propably output index file.
}
```

Reply:
I like the idea! 👍

Will play with that later

I think it's a good thing to not rush through this change, so I think this PR will be active some time with some small changes now and then...

Reply:
I played a bit with this, and this approach would break the SettingsMap and SettingsLoader logic... :/
```
  settingsMap.forEach((propertyName, settingName) => {
    const configuredValue = config.get(settingName);
    if (configuredValue !== undefined) {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (settings as any)[propertyName] = configuredValue;
    }
  });
  ```

Reply:
Of the top of my head and a bit naive perhaps but would this work? We could also rebuild the settingsmap I guess?
```javascript
 settingsMap.forEach((propertyName, settingName) => {
     // settingName.split('.')[0] would be "NAB" (?)
     const groupName = settingName.split('.')[1];
     groupName = settingName.split('.')[2];
    const configuredValue = config.get(settingName);
    if (configuredValue !== undefined) {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (settings as any)[groupName][propertyName] = configuredValue;
    }
  });
```

Reply:
You are welcome to test that out! 🙂